command/agent/agent.go

```diff
@@ -69,14 +73,13 @@ func NewAgent(config *Config, logOutput io.Writer) (*Agent, error) {
 	if a.client == nil && a.server == nil {
 		return nil, fmt.Errorf("must have at least client or server mode enabled")
 	}
-	if a.config.ConsulConfig.AutoRegister {
-		if err := a.syncAgentServicesWithConsul(a.serverHTTPAddr, a.clientHTTPAddr); err != nil {
+	if a.config.Consul.AutoRegister {
+		if err := a.syncAgentServicesWithConsul(); err != nil {
 			a.logger.Printf("[ERR] agent: unable to sync agent services with consul: %v", err)
 		}
-		if a.consulService != nil {
-			go a.consulService.PeriodicSync()
-		}
 	}
+	go a.consulSyncer.Run()
```

Comment:
@sean- I think the UX is better if we hide both the options behind the same flag, but currently the auto-registration feature doesn't allow operators to specify tags or custom checks for the services so until we are confident that users won't need to take care of the registration out of band I think we should keep these two flags separate.